File: src/main/java/lechuck/intellij/TaskRunConfigurationType.java
package lechuck.intellij;

import com.intellij.execution.configurations.ConfigurationTypeBase;

public class TaskRunConfigurationType extends ConfigurationTypeBase {
    public static final String ID = "TaskRunConfiguration";

    public TaskRunConfigurationType() {
        super(ID, "Taskfile", "Taskfile run configuration type", TaskPluginIcons.Task);
        addFactory(new TaskConfigurationFactory(this));
    }
}

File: src/main/kotlin/lechuck/intellij/TaskRunConfigurationType.kt
package lechuck.intellij

import com.intellij.execution.configurations.ConfigurationTypeBase

class TaskRunConfigurationType :
    ConfigurationTypeBase(ID, "Taskfile", "Taskfile run configuration type", TaskPluginIcons.Task) {
    companion object {
        const val ID = "TaskRunConfiguration"
    }

    init {
        addFactory(TaskConfigurationFactory(this))
    }
}